### PR TITLE
make sure that we always use the correct owner for both source and target

### LINKED ISSUE #100: lib/oc-tests/test_sharePermissions.py test is failing with encryption
[https://ci.owncloud.org/job/smashbox/22](https://ci.owncloud.org/job/smashbox/22/TEST_NAME=test_sharePermissions,compiler=g++,label=master,mirallBranch=v1.8.1,qt=qt4,serverVersion=owncloud8.1/console)
- ownCloud: master
- client: 1.8.1

```
$ bin/smash lib/oc-tests/test_sharePermissions.py -t=0
SUMMARY:smash.:runid: 150602-161105
SUMMARY:smash.:URL: http://test:***@127.0.0.1/ownCloud/Smashbox/master/remote.php/webdav/
SUMMARY:smash.:Running iteration 1
SUMMARY:smash.:running lib/oc-tests/test_sharePermissions.py in /home/nickv/ownCloud/Smashbox/tmpdir/test_sharePermissions-150602-161105 as test testset #0 None
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
2015-06-02 16:11:21,143 - ERROR - recipient_worker - Got status code 500
2015-06-02 16:11:21,165 - ERROR - recipient_worker - Operation "move_in_overwrite" must be allowed when share permissions are 31 'Operation "%s" must be %s when share permissions are %i' % (operation, success_message, perms) failed in recipient_worker() ["/home/nickv/ownCloud/Smashbox/smashbox/lib/oc-tests/test_sharePermissions.py" at line 251]
INFO:requests.packages.urllib3.connectionpool:Resetting dropped connection: 127.0.0.1
2015-06-02 16:11:22,465 - ERROR - recipient_worker - Got status code 500
2015-06-02 16:11:22,466 - ERROR - recipient_worker - Operation "move_in_subdir_overwrite" must be allowed when share permissions are 31 'Operation "%s" must be %s when share permissions are %i' % (operation, success_message, perms) failed in recipient_worker() ["/home/nickv/ownCloud/Smashbox/smashbox/lib/oc-tests/test_sharePermissions.py" at line 251]
INFO:requests.packages.urllib3.connectionpool:Resetting dropped connection: 127.0.0.1
2015-06-02 16:11:24,227 - ERROR - recipient_worker - Got status code 500
2015-06-02 16:11:24,228 - ERROR - recipient_worker - Operation "delete" must be allowed when share permissions are 31 'Operation "%s" must be %s when share permissions are %i' % (operation, success_message, perms) failed in recipient_worker() ["/home/nickv/ownCloud/Smashbox/smashbox/lib/oc-tests/test_sharePermissions.py" at line 251]
INFO:requests.packages.urllib3.connectionpool:Resetting dropped connection: 127.0.0.1
2015-06-02 16:11:25,259 - ERROR - recipient_worker - Got status code 500
2015-06-02 16:11:25,260 - ERROR - recipient_worker - Operation "rmdir" must be allowed when share permissions are 31 'Operation "%s" must be %s when share permissions are %i' % (operation, success_message, perms) failed in recipient_worker() ["/home/nickv/ownCloud/Smashbox/smashbox/lib/oc-tests/test_sharePermissions.py" at line 251]
2015-06-02 16:11:26,294 - ERROR - recipient_worker - 4 error(s) reported
CRITICAL:smash.:Aborting run -- non-zero exit code (2)
```

Most likely caused by changes in master last week, because it worked before:
- [ ] :x: `bin/smash lib/oc-tests/test_sharePermissions.py -t=0`: Permission: all
- [x] `bin/smash lib/oc-tests/test_sharePermissions.py -t=1`: Permission: read
- [x] `bin/smash lib/oc-tests/test_sharePermissions.py -t=2`: Permission: read + create
- [x] `bin/smash lib/oc-tests/test_sharePermissions.py -t=3`: Permission: read + update
- [ ] :x: `bin/smash lib/oc-tests/test_sharePermissions.py -t=4`: Permission: read + delete
- [x] `bin/smash lib/oc-tests/test_sharePermissions.py -t=5`: Permission: read + create + update
- [ ] :x: `bin/smash lib/oc-tests/test_sharePermissions.py -t=6`: Permission: read + create + delete
- [ ] :x: `bin/smash lib/oc-tests/test_sharePermissions.py -t=7`: Permission: read + update + delete

So it looks like delete is the broken permission...

### LINKED ISSUE #16704: Deleting a file as share receiver does not copy enc keys
### Steps to reproduce
1. Share a folder
2. Create a file
3. As share recipient delete the file

or

`bin/smash -t=4 lib/oc-tests/test_sharePermissions.py` see https://github.com/owncloud/smashbox/issues/100

### Expected behaviour
1. The file and owner keys are in `owner/files_encryption/keys/files_trashbin/files/filename.d<timestamp>`
2. The share receiver key is in `receiver/files_encryption/keys/files_trashbin/files/filename.d<timestamp>`

### Actual
1. :white_check_mark: 
2. :x:

This means if the receiver restores the file, it is going to be empty.

@schiesbn

### PR DESCRIPTION
**Pull request by schiessle**
make sure that we always use the correct owner for both source and target

fix #16704 
fix owncloud/smashbox#100

cc @nickvergessen

### CODE REVIEW

**Comment by nickvergessen** [comment]
Tested via smashbox, makes `lib/oc-tests/test_sharePermissions.py` pass again:
:+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by schiesbn** [comment]
don't merge if Jenkins finished. I want to push a additional unit test first, thanks!

**Comment by MorrisJobke** [comment]
@schiesbn Okay :)

**Comment by PVince81** [comment]
Tested and works :+1:

File: tests/lib/encryption/keys/storage.php [hunk 1]
```diff
@@ -361,7 +361,46 @@ public function dataProviderCopyRename() {
...
+
+		$this->util->expects($this->any())
+			->method('getUidAndFilename')
+			->will($this->returnCallback(array($this, 'getUidAndFilenameCallback')));
+		$this->util->expects($this->any())
+			->method('isSystemWideMountPoint')
+			->willReturn($systemWideMountPoint);
+
+		$this->assertSame($expected,
+			\Test_Helper::invokePrivate($this->storage, 'getPathToKeys', [$path])
```
Comment (nickvergessen) [inline tests/lib/encryption/keys/storage.php]: deprecated :stuck_out_tongue_winking_eye: https://github.com/owncloud/core/pull/16710

**Comment by nickvergessen** [comment]
@MorrisJobke there was an open comment?

**Comment by DeepDiver1975** [comment]
> don't merge if Jenkins finished. I want to push a additional unit test first, thanks!

> deprecated :stuck_out_tongue_winking_eye: #16710

@MorrisJobke you are too fast

**Comment by MorrisJobke** [comment]
> @MorrisJobke you are too fast

Damnit ... I've overseen that :(

**Comment by MorrisJobke** [comment]
I will prepare a PR

**Comment by MorrisJobke** [comment]
#16719

**Pull request merged**